### PR TITLE
fix: GetFormDefinitionFieldsHandler logic

### LINKED ISSUE #746: Bug: GET forms/{formId}/definition/fields returns 500 for forms with empty definition
### Goal

`GET /api/forms/{formId}/definition/fields` should return a successful response (typically **200** with a JSON array, possibly empty) for valid forms so clients can build submission grids with dynamic answer columns. Today it can return **500** `application/problem+json`, which blocks correct UX.

### Observed

- Hub calls `GET https://{api}/api/forms/{formId}/definition/fields` in parallel with listing submissions.
- Form has an empty definition e.g. JsonData is `"{ }"`
- For at least some forms (e.g. new or without submissions in local repro), the endpoint responds **500** in a few milliseconds.
- Response content type: `application/problem+json`.
- Stack
```bash
TYPE: InvalidOperationException
REASON: The requested operation requires an element of type 'Object', but the target element has type 'String'.
---------------------------------
   at System.Text.Json.ThrowHelper.ThrowJsonElementWrongTypeException(JsonTokenType expectedType, JsonTokenType actualType)
   at System.Text.Json.JsonDocument.TryGetNamedPropertyValue(Int32 index, ReadOnlySpan`1 propertyName, JsonElement& value)
   at Endatix.Core.UseCases.FormDefinitions.GetFields.GetFormDefinitionFieldsHandler.Handle(GetFormDefinitionFieldsQuery request, CancellationToken cancellationToken) in /Users/johnnybravo/things/src/Endatix.Core/UseCases/FormDefinitions/GetFields/GetFormDefinitionFieldsHandler.cs:line 59
```

### Expected

- **200** with `[]` or the field list when the form exists and the caller is authorized.
- **404** (or consistent not-found contract) when the form does not exist — not 500 for normal edge cases.

### Hub mitigation (tracked separately)

Hub now degrades when this call fails: it still renders the submissions page with **system columns only** and a non-blocking status message, so users see the empty/list state instead of a hard error. Fixing the API remains the source-of-truth fix for dynamic columns.

### Notes

- Hub path: `GET /forms/{formId}/definition/fields` (see `Definitions.getFields` in Hub client).
- Please add or extend integration coverage for this route for empty / minimal-definition forms.

### PR DESCRIPTION
**Pull request by PeterMarinov**
# fix: GetFormDefinitionFieldsHandler logic

## Description
- Ensure GetFields endpoint returns 200 when there are no definitions or definition is empty
- Add `ByIdReadOnly` specification to FormSpecifications
- Improve GetFormDefinitionFieldsHandler logic for safe JSON parsing and extraction. Add support for panel dynamic and matrix, dynamic matrix questions
- Update tests

## Related Issues
- closes https://github.com/endatix/endatix/issues/746

## Type of Change
Check all that apply.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Dependency update
- [x] Refactoring (non-breaking change which improves code quality)
- [ ] Other (please describe)

## Checklist
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

> [!NOTE]
> - [x] I confirm that my Pull Request follows all the checklist requirements above

## Screenshots
image

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected HTTP 404 error message for missing forms in the form definitions endpoint.
  * Improved JSON parsing with enhanced error handling for malformed data.

* **New Features**
  * Expanded field extraction to support additional form container types (dynamic panels, matrix dropdowns, matrix dynamic).
  * Enhanced field metadata by using element titles when available.

* **Tests**
  * Added comprehensive test coverage for JSON parsing edge cases and various form structure types.
  * Updated test scenarios to reflect modified form validation behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/endatix/endatix/pull/747)

### CODE REVIEW

File: src/Endatix.Core/UseCases/FormDefinitions/GetFields/GetFormDefinitionFieldsHandler.cs [hunk 1]
```diff
@@ -75,50 +128,140 @@ private static void ExtractFields(JsonElement elements, Dictionary<string, (stri
...
+            }
+
+            if (!TryGetObjectProperty(column, "name", out var nameProp))
+            {
+                continue;
+            }
+
+            var nameValue = nameProp.GetString();
+            if (string.IsNullOrWhiteSpace(nameValue))
+            {
```
[the remark this answers is not in the document]
Comment (PeterMarinov) [reply src/Endatix.Core/UseCases/FormDefinitions/GetFields/GetFormDefinitionFieldsHandler.cs]: Done. Please review
